packages/cli/src/modules/insights/database/repositories/insights-by-period.repository.ts

```diff
@@ -309,16 +310,29 @@ export class InsightsByPeriodRepository extends Repository<InsightsByPeriod> {
 				'period',
 			)
 			.addSelect('insights.type', 'type')
-			.addSelect('SUM(value)', 'total_value')
-			// Use a cross join with the CTE
-			.innerJoin('date_ranges', 'date_ranges', '1=1')
-			// Filter to only include data from the last 14 days
-			.where('insights.periodStart >= date_ranges.previous_start')
+			.addSelect('SUM(value)', 'total_value');
+
+		if (projectId) {
+			rawRowsQuery
+				.innerJoin('insights.metadata', 'metadata')
+				// Use a cross join with the CTE
+				.innerJoin('date_ranges', 'date_ranges', '1=1')
+				.where('insights.periodStart >= date_ranges.previous_start')
+				.andWhere('metadata.projectId = :projectId', { projectId });
+		} else {
+			rawRowsQuery
+				// Use a cross join with the CTE
+				.innerJoin('date_ranges', 'date_ranges', '1=1')
+				.where('insights.periodStart >= date_ranges.previous_start');
+		}
```

Comment:
```suggestion
			.addSelect('SUM(value)', 'total_value')
			// Use a cross join with the CTE
			.innerJoin('date_ranges', 'date_ranges', '1=1')
			.where('insights.periodStart >= date_ranges.previous_start');

		if (projectId) {
			rawRowsQuery
				.innerJoin('insights.metadata', 'metadata')
				.andWhere('metadata.projectId = :projectId', { projectId });
		}
```

wouldn't this be the same and we can avoid the if /else branching ? just asking

Reply:
Yup that's much better, and result in the same query 👍